tests/api2/jail_plugins.py

```diff
@@ -0,0 +1,291 @@
+
+import pytest
+import sys
+import os
+from time import sleep
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from auto_config import pool_name
+from functions import GET, POST, DELETE
+
+JOB_ID = None
+job_results = None
+not_freenas = GET("/system/is_freenas/").json() is False
+reason = "System is not FreeNAS skip Jails test"
+to_skip = pytest.mark.skipif(not_freenas, reason=reason)
+plugin_repos = 'https://github.com/freenas/iocage-ix-plugins.git'
+
+plugins_list = [
+    'asigra',
+    'backuppc',
+    'bacula-server',
+    'bru-server',
+    'clamav',
+    'couchpotato',
+    'deluge',
+    'emby',
+    'gitlab',
+    'irssi ',
+    'jenkins',
+    'jenkins-lts',
+    'madsonic',
+    'mineos',
+    'nextcloud',
+    'plexmediaserver',
+    'plexmediaserver-plexpass',
+    'qbittorrent',
+    'quasselcore',
+    'radarr',
+    'redmine',
+    'rslsync',
+    'sonarr',
+    'subsonic',
+    'syncthing',
+    'tarsnap',
+    'transmission',
+    'weechat',
+    'xmrig',
+    'zoneminder'
+]
+
+plugins_objects = [
+    "id",
+    "state",
+    "type",
+    "release",
+    "plugin_repository"
+]
+
+
+@to_skip
+def test_01_activate_jail_pool():
+    results = POST('/jail/activate/', pool_name)
+    assert results.status_code == 200, results.text
+    assert results.json() is True, results.text
+
+
+@to_skip
+def test_02_verify_jail_pool():
+    results = GET('/jail/get_activated_pool/')
+    assert results.status_code == 200, results.text
+    assert results.json() == pool_name, results.text
+
+
+@to_skip
+def test_03_get_list_of_installed_plugin():
+    results = GET('/plugin/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list), results.text
+
+
+@to_skip
+def test_04_verify_plugin_repos_is_in_official_repositories():
+    results = GET('/plugin/official_repositories/')
+    assert plugin_repos in results.json(), results.text
+
+
+@to_skip
+def test_05_get_list_of_available_plugins_job_id():
+    global JOB_ID
+    payload = {
+        "plugin_repository": plugin_repos
+    }
+    results = POST('/plugin/available/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), int), results.text
+    JOB_ID = results.json()
+
+
+@to_skip
+def test_06_verify_list_of_available_plugins_job_id_is_successfull():
+    global job_results
+    while True:
+        job_results = GET(f'/core/get_jobs/?id={JOB_ID}')
+        job_state = job_results.json()[0]['state']
+        if job_state in ('RUNNING', 'WAITING'):
+            sleep(3)
+        else:
+            assert job_state == 'SUCCESS', str(job_results)
+            break
+
+
+@to_skip
+@pytest.mark.parametrize('plugin', plugins_list)
+def test_07_verify_available_plugin_(plugin):
+    for plugin_info in job_results.json()[0]['result']:
+        if plugin in plugin_info:
```

Comment:
This would need updating as changes are introduced in the plugin list. How about we come up with a more robust solution getting contents ( `requests` module ) of `https://raw.githubusercontent.com/freenas/iocage-ix-plugins/master/INDEX` and then converting to json and verifying list of plugins there and the available returned ?

Reply:
What about this https://github.com/freenas/freenas/pull/3301/commits/133d0a0cc9f73490b002aebfd433d7047e016c2d

Reply:
@ericbsd it's much better, thank you ;)